### PR TITLE
UITOOL-210/plasma thumbnail for overview page

### PR DESCRIPTION
**Pull request by aiheon**
### Proposed Changes

https://coveord.atlassian.net/browse/UITOOL-210

Created the new Tile Component that will be used in the Home page of Plasma :tada: 

image

### Potential Breaking Changes

None

### Acceptance Criteria

-   [X] The proposed changes are covered by unit tests
-   [X] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by aiheon** [comment]
image

**Comment by aiheon** [comment]
image

**Comment by j-mcgregor** [comment]
Not sure if this is something you have any control over but there's a weird fuzzing effect (screen rec doesn't show it well) when you hover over the edge. The transitions is much smoother though, nice work

https://user-images.githubusercontent.com/22773767/143601492-96977282-19ba-4c70-8754-690acb242f33.mov

File: packages/vapor/scss/components/tile.scss [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    position: absolute;
+    bottom: 0.25px;
+    width: 100%;
+    height: 51.75px;
+    padding: 12px 14px;
+
+    overflow: hidden;
+    background: var(--white);
+    border-radius: 8px;
+    transition: height 300ms;
```
Comment (j-mcgregor) [inline packages/vapor/scss/components/tile.scss:42]: nice

**Review by j-mcgregor — APPROVED** [review]

**Comment by aiheon** [comment]
> Not sure if this is something you have any control over but there's a weird fuzzing effect (screen rec doesn't show it well) when you hover over the edge. The transitions is much smoother though, nice work
> 
>  Screen.Recording.2021-11-26.at.10.15.33.AM.mov

It does that only from the bottom border :thinking: Probably something to do with the double animation for the drawer and the tile moving up on hover :thinking:

File: packages/vapor/scss/components/tile.scss (hunk 1 above)
Comment (aiheon) [reply packages/vapor/scss/components/tile.scss:42]: :smile:

File: packages/vapor/scss/components/tile.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    position: relative;
+    top: 20px;
+    left: 20px;
+    width: 264px;
+    height: auto;
+    min-height: 221px;
+    background: var(--white);
+    border: 1px solid var(--grey-40);
+    border-radius: 8px;
+    box-shadow: 0px 2px 8px rgba(229, 232, 232, 0.75);
```
Comment (gdostie) [inline packages/vapor/scss/components/tile.scss:11]: Any css variable we can reuse from [here](https://github.com/coveo/react-vapor/blob/master/packages/vapor/scss/utility/shadow.scss#L2-L10)?

File: packages/vapor/scss/components/tile.scss [hunk 3]
```diff
@@ -0,0 +1,65 @@
+.tile {
+    position: relative;
+    top: 20px;
+    left: 20px;
+    width: 264px;
+    height: auto;
```
Comment (gdostie) [inline packages/vapor/scss/components/tile.scss:6]: What is this one for?

**Review by gdostie — COMMENTED** [review]
I get Jack's glitch too

File: packages/vapor/scss/components/tile.scss (hunk 2 above)
Comment (aiheon) [reply packages/vapor/scss/components/tile.scss:11]: Oh maybe! :eyes:

File: packages/vapor/scss/components/tile.scss (hunk 3 above)
Comment (aiheon) [reply packages/vapor/scss/components/tile.scss:6]: the height? :eyes: 
I can try and remove it to see if nothing breaks :+1:

**Comment by aiheon** [comment]
> I get Jack's glitch too

It's the transform: translate3d that is doing that :female_detective:  ... i'll just remove it XD

**Review by gdostie — APPROVED** [review]
Is the tile component responsive? I see a lot of fixed dimensions in pixels and it has me worried about its ability to grow with the screen width

**Comment by aiheon** [comment]
> Is the tile component responsive? I see a lot of fixed dimensions in pixels and it has me worried about its ability to grow with the screen width

Yeah it stays small :monocle_face: 
I'll test with min-width min-height etc :female_detective: 
image

**Comment by dmgauthier** [comment]
IDK if you find that problematic or expected but the puzzle piece is slightly moving on hover, I guess it will even with "real stuff". Personnally I don't mind. It's looking nice.

**Review by dmgauthier — APPROVED** [review]

**Comment by aiheon** [comment]
> IDK if you find that problematic or expected but the puzzle piece is slightly moving on hover, I guess it will even with "real stuff". Personnally I don't mind. It's looking nice.

Yes, the change of color for the borders on hover is doing that. I also don't mind it and when I tried to force the tile to go up to camouflage this effect, I created the nice glitch you see in Jack's comment :see_no_evil:

File: packages/demo/src/plasma/Home.tsx [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import '@demo-styling/home.scss';
 
 import React from 'react';
+
 import {Section} from 'react-vapor';
+import {Tile} from 'react-vapor';
```
Comment (rosalie-liu) [inline packages/demo/src/plasma/Home.tsx:6]:
```suggestion
import {Section, Tile} from 'react-vapor';
```

File: packages/demo/src/plasma/Home.tsx [hunk 5]
```diff
@@ -17,56 +19,234 @@ const Home: React.FC = () => (
         <Section className="section">
             <h2>Fondations</h2>
```
Comment (rosalie-liu) [inline packages/demo/src/plasma/Home.tsx:20]:
```suggestion
            <h2>Foundations</h2>
```

File: packages/demo/src/plasma/Home.tsx [hunk 6]
```diff
@@ -17,56 +19,234 @@ const Home: React.FC = () => (
...
+                <Tile
+                    title="Patate King"
+                    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec vel justo sit amet tortor aliquam. Patate King is a God. 🥔"
+                    svgName="plasmaComponentBox"
+                    href="#/styles/icons/list"
+                />
+            </div>
         </Section>
         <Section className="section">
             <h2>Disply & Utilities</h2>
```
Comment (rosalie-liu) [inline packages/demo/src/plasma/Home.tsx:228]:
```suggestion
            <h2>Display & Utilities</h2>
```

File: packages/vapor/scss/components/tile.scss [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        border: $border-radius solid var(--digital-blue-60);
+        box-shadow: var(--medium-elevation-on-light);
+    }
+}
+
+.tile-icon {
+    position: relative;
+    height: 173px;
+
+    & svg {
```
Comment (rosalie-liu) [inline packages/vapor/scss/components/tile.scss:21]:
```suggestion
    svg {
```

File: packages/demo/src/plasma/Home.tsx (hunk 4 above)
Comment (aiheon) [reply packages/demo/src/plasma/Home.tsx:6]: Senpai!! :heart:

**Review by sallainCoveo — APPROVED** [review]

**Pull request merged**